src/main/java/com/redhat/cloud/notifications/migration/policynotification/PoliciesMigrationService.java

```diff
@@ -0,0 +1,103 @@
+
+    @Inject
+    ApplicationResources applicationResources;
+
+    static class MigrateResponse {
+        public AtomicInteger eventTypesMigrated = new AtomicInteger();
+        public AtomicInteger accountsMigrated = new AtomicInteger();
+    }
+
+    @Path("/migrate")
+    @GET
+    public Uni<MigrateResponse> migrate() {
+        final MigrateResponse response = new MigrateResponse();
+        return applicationResources.getEventType(BUNDLE, APPLICATION, EVENT_TYPE)
+        .onItem().transformToMulti(eventType -> {
+            return policyNotifications.getSubscriptions()
+                    .onItem().transformToMulti(policiesEmailSubscriptions -> Multi.createFrom().iterable(policiesEmailSubscriptions))
+                    .onItem().transformToMulti(pes -> emailSubscriptionResources.subscribe(
+                    pes.accountId,
+                    pes.userId,
+                    BUNDLE,
+                    APPLICATION,
+                    fromPoliciesEventType(pes.eventType.toLowerCase())
+            )
+                    .invoke(() -> response.eventTypesMigrated.incrementAndGet())
+                    .onItem().transform(subscribed -> pes.accountId).toMulti())
+                    .concatenate().collectItems().in(HashSet::new, HashSet::add)
+                    .onItem().transformToMulti(accountIds -> Multi.createFrom().iterable(accountIds))
+                    .onItem().castTo(String.class)
+                    .onItem().transformToMultiAndConcatenate(accountId -> {
+                        response.accountsMigrated.incrementAndGet();
+                        // Remove existing endpoints
+                        return endpointResources.getLinkedEndpoints(accountId, eventType.getId(), new Query())
+                        .onItem().transformToMultiAndConcatenate(endpoint -> endpointResources.unlinkEndpoint(accountId, endpoint.getId(), eventType.getId()).toMulti())
+                        // Create EmailAction (Endpoint) and add it to the eventType
+                        .collectItems().asList()
+                        .onItem().transformToUni(_removed -> {
+                            Endpoint endpoint = new Endpoint();
+                            endpoint.setTenant(accountId);
+                            endpoint.setEnabled(true);
+                            endpoint.setDescription("");
+                            endpoint.setName("email-endpoint");
+                            endpoint.setType(EndpointType.EMAIL_SUBSCRIPTION);
+                            endpoint.setProperties(new EmailSubscriptionAttributes());
+                            return endpointResources.createEndpoint(endpoint);
+                        })
+                        .onItem().transformToUni(endpoint -> endpointResources.linkEndpoint(accountId, endpoint.getId(), eventType.getId())).toMulti();
+                    });
+        }).collectItems().asList().onItem().transform(objects -> response);
+    }
```

Comment:
What happens if this is interrupted halfway for any reason? This looks indempotent (and therefore safe) to me but I'd like you to confirm I didn't miss anything.

Reply:
Running again should yield the desired results, but we might have partial results (Assuming this is not a single transaction). This does not delete our "inputs".

Reply:
Well it does not delete the email subscriptions but we're safe because of the `INSERT INTO ... ON CONFLICT ... DO NOTHING`.

Reply:
Yes, those are not supposed to fail if it already exists (or if it doesn't exist in the case of removing).